# US Powerball Winning Numbers Analysis

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('data/powerball_winning_numbers.csv')
df.head()

Draw Date    Winning Numbers  Multiplier
0  02/03/2010  17 22 36 37 52 24         2.0
1  02/06/2010  14 22 52 54 59 04         3.0
2  02/10/2010  05 08 29 37 38 34         5.0
3  02/13/2010  10 14 30 40 51 01         4.0
4  02/17/2010  07 08 19 26 36 15         3.0

In [3]:
df.tail()

Draw Date    Winning Numbers  Multiplier
1016  11/02/2019  03 23 32 37 58 22         2.0
1017  11/06/2019  15 28 46 62 64 17         3.0
1018  11/09/2019  14 17 35 38 60 25         2.0
1019  11/13/2019  23 26 27 28 66 11         2.0
1020  11/16/2019  14 22 26 55 63 26         3.0

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1021 entries, 0 to 1020
Data columns (total 3 columns):
Draw Date          1021 non-null object
Winning Numbers    1021 non-null object
Multiplier         811 non-null float64
dtypes: float64(1), object(2)
memory usage: 24.0+ KB


In [6]:
df.isna().sum()

Draw Date            0
Winning Numbers      0
Multiplier         210
dtype: int64

In [5]:
df['Winning Numbers'][:10]

0    17 22 36 37 52 24
1    14 22 52 54 59 04
2    05 08 29 37 38 34
3    10 14 30 40 51 01
4    07 08 19 26 36 15
5    13 27 37 41 54 32
6    04 17 35 50 57 12
7    18 47 51 53 58 30
8    07 09 14 45 49 23
9    10 29 33 41 59 15
Name: Winning Numbers, dtype: object

**Initial Observation:**
- Looks like there are 210 drawings without Multipliers
- Winning Numbers column contains the all the numbers seperated by a space